mantidproject/mantid
# Event filtering documentation

The documentation wasn't clear and didn't actually match the implementation. This is a rework of much of the documentation to make it match the code. The main feature is giving code examples of creating a custom event filter in python.

There was also a small amount of work modernizing bits of the code and adding parameter validation to improve error reporting to the user.

**To test:**

Build the docs and see if they are better than before.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
